### DISCUSSION TITLE
URL CONFIGURATION ISSUE

### DISCUSSION THREAD

**Discussion by HelpiScribe** [supabase/supabase#31617] [Questions]
I have been having a Login issue in the last 24 hours as follows:
1. When I tried to login is popup "Invalid API Key
2. I asked on both the Development chat and Lovable assistant GPT,
they both said I should fix:

Go to your Supabase project settings
Navigate to Authentication > URL Configuration
Configure the following:
Site URL should be set to your application's base URL. For
development, this would typically be:

http://localhost:5173/ (if you're using Vite's default port)
Redirect URLs should include:

http://localhost:5173/** (for development)
Any additional production URLs if you have them

I did all that at, but the error still persists; see the attached
screenshot for reference.

Here is the app I'm working on:
https://lovable.dev/projects/c1118834-064f-40a7-87e3-7cd085f1c973
 URL: https://academic-career.lovable.app/
GitHub: https://github.com/HelpiScribe/academic-career

Custom Domain deploy: https://www.helpiscribe.com/

Please help on how to resolve this, as you always do.

Warm regards,
Emmanuel
![URL Config Issue](https://github.com/user-attachments/assets/918d1d31-b832-4447-a5f6-9fd8152cda5f)

**Comment by bytaesu** [comment]
1. If it’s a production state, ensure that the environment variables are correctly configured. (Based on the site you provided, it seems to be deployed on Vercel. I recommend using the Vercel integration available on the Supabase dashboard.)

2. Use the wildcard for the deployed site as well. (e.g. " https://www.helpiscribe.com/** ", " https://academic-career.lovable.app/** ", " https://helpiscribe.vercel.app/** "  ...)

3. For the Site URL, set the deployed domain, and for the development environment, handle it by adding the redirect URL.

I don’t think the “Invalid API Key” error is caused by the Redirect URL configuration. In my personal opinion, please check (1).

**Reply by HelpiScribe** [reply]
@bytaesu, thank you very much for your prompt response. Please I'm new to this, Lovable.Dev just makes it easy for my idea to come out. Now, could you tell me:    
                                                                                                                                                                                       
Site URL: https://www.helpiscribe.com/**                                                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                       
Redirect URLs:  
(1) https://academic-career.lovable.app/**                                                                                                                      
(2) https://helpiscribe.vercel.app/** 
Is that right?

**Reply by bytaesu** [reply]
@HelpiScribe 

As explained in the dashboard, the Site URL should be the base address of your website (e.g., https://supabase.com). If none of the redirect URLs are valid, the user will be redirected to the Site URL. Wildcards are not allowed. For more details, please refer to the official documentation: https://supabase.com/docs/guides/auth/redirect-urls.

Example:

**Site URL:** https://www.helpiscribe.com

**Redirect URLs:**
• https://www.helpiscribe.com/**
• https://helpiscribe.vercel.app/**
• https://academic-career.lovable.app/**
• http://localhost:5173/**
• https://other-your-server-url/**

…

As mentioned above, the “Invalid API Key” error is likely caused by an issue with injecting the API key when initializing the Supabase client. This is especially important in production deployments, where environment variables need to be correctly configured.
(Even if the development environment works well, it does not “automatically” apply to the production environment.)

**Reply by HelpiScribe** [reply]
Oh wow! Thank so much. on the "... injecting the API key when initializing the Supabase client." what do you advice I should do to remedy this?

Thanks

**Reply by bytaesu** [reply]
I was simply referring to just inserting the supabaseURL and supabaseKEY like this.

<img width="521" alt="img" src="https://github.com/user-attachments/assets/82485a39-d430-4005-9718-0f95357a2462" />

**Reply by HelpiScribe** [reply]
Thanks for your support. 

I config the URL and Redirect URL but same issue when trying to login: 
![image](https://github.com/user-attachments/assets/87b1fbe4-3a92-475f-8a10-7fd530201cd9)
![image](https://github.com/user-attachments/assets/a573f963-0133-4c9d-ab52-f5dd25439f12)

**Reply by bytaesu** [reply]
1. Check if you are using the environment variables correctly.
2. Check if there are any typos in the environment variables.
3. Check if you are using the correct Supabase URL and key values.
4. Do not use examples directly. (Make your own judgment based on your product.)

**Reply by HelpiScribe** [reply]
I have followed all your instructions/guide. This is what I get:
![image](https://github.com/user-attachments/assets/67bac4f5-e8e9-4ae6-8e71-4d0c29700119)

I don't know what to do again.

**Reply by bytaesu** [reply]
It doesn’t seem to be a Supabase Auth error message. 
This time, I won’t be able to assist you. It’s difficult to provide answers for every situation.

**Reply by HelpiScribe** [reply]
Thanks for your support! Lovable team told me to disconnect and reconnect, also add up to your instruction. Now its working fine. You can check it out. I truly appreciate your promptness!

**Reply by bytaesu** [reply]
That's good news 🙂
